src/MoBi.Presentation/DTO/MoleculeSelectionDTO.cs

```diff
@@ -16,13 +16,15 @@ public MoleculeSelectionDTO()
          Rules.Add(AllRules.SelectedMoleculesHaveUniqueNames);
       }
 
-      public string BuildingBlock { get; set; }
+      public IBuildingBlock BuildingBlock => MoleculeBuilder.BuildingBlock;
```

Comment:
Line 22 we are checking for MolecleBuilder defined. But here you are not. Can the MoleculeBuilder be null?